software-mansion/react-native-reanimated
# Add more tests - useSharedValue, useAnimatedStyle , useDerivedValue

## Summary
**The new test runs 35 seconds ⏰**

In this PR I add two tests of our core functionalities:
* useSharedValue
* useAnimatedStyle - test reusing animated style to animate several components simultaneusly
* useDerivedValue - test some basic snapshots
* useDerivedValue - test chaining 10 derived values

I also
* Mark Presets as not suitable for spell-check
* Modify color function to JSON.stringify objects
* Add additional types `null | undefined` to TestValue type so, after some more checks we will be able to remove function `expectToBeNullable`
* Create function `getComparisonModeForProp`

<img width="542" alt="image" src="https://github.com/software-mansion/react-native-reanimated/assets/56199675/6de4c38e-f1c9-4815-9dbc-59b435dbb26e">

## Recording


https://github.com/software-mansion/react-native-reanimated/assets/56199675/dcd11c06-b9a5-454c-be30-e56999d84497
